fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/api/event/client/ClientJoinWorldCallback.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.client;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.client.world.ClientWorld;
+
+/**
+ * A callback for when the client player joins a new world.
+ */
+public interface ClientJoinWorldCallback {
```

Comment:
Why this over something when a player joins the world that is fired on both side?

Reply:
This is client loading a world for rendering. the player is not set up at the point the world is ready.

Reply:
Removed this event, guess no one cares about client loading a new client world